src/lib/OpenEXR/ImfInputFile.h

```diff
@@ -118,8 +110,8 @@ class IMF_EXPORT_TYPE InputFile : public GenericInputFile
     // Access to the file header
     //--------------------------
 
-    OPENEXR_DEPRECATED (
-        "Use context-based attribute access for faster retrieval")
+    //OPENEXR_DEPRECATED (
```

Comment:
the deprecation attribute is SUPER annoying because -Wall causes SO much spam. What if we added to the cmake, "EXR_WARN_ON_DEPRECATED=OFF" and predicate the OPENEXR_DEPRECATED macro on that? deprecation warnings are useful, but I'd only want to turn them on briefly when migrating to a new version of the library

Reply:
yeah, not 100% sure what to do with that, I feel like I want to put them in there prior to release, but you're right, we should think about a convenient enough way for a dev. to silence the deprecations if they can't change things just yet...

Reply:
That's easy to do in the downstream project, it's a specific warning

Reply:
But we could also have a magic symbol that if set before inclusion of the exr headers, just defines all the DEPRECATED macros to do nothing.

Reply:
The thing about setting it downstream is that that becomes all or nothing; you might want your own deprecation warnings, but not the ones out of your control. For this reason, it's a fairly usual pattern to allow suppression of a project's deprecations via a macro.

Reply:
That's what I meant. If you set OPENEXR_NO_DEPRECATION_WARNINGS prior to including the openexr headers, you won't get the warnings. (I mean, we'd need to honor that.)